client/src/pages/Property.tsx

```diff
@@ -17,8 +17,12 @@ const API_URL = `${import.meta.env.VITE_API_SERVER_URL}/api/v1`;
  *
  */
 const Property = () => {
-  const { state } = useLocation();
+  const { id } = useParams();
   const toast = useToast();
+  const [post, setPost] = useState<Post | null>(null);
+  const [loadingPost, setLoadingPost] = useState<null | boolean>(null);
```

Comment:
Since you added the `if post` check below just now, `loadingPost` can be just `boolean` default to `false`.